roledb.go

```diff
@@ -335,6 +342,27 @@ func (x *sqlGroupDB) GetGroups() ([]*AuthGroup, error) {
 	return readAllGroups(x.db.Query("SELECT id,name,permlist FROM authgroup"))
 }
 
+func (x *sqlGroupDB) GetGroupsRaw() ([]RawAuthGroup, error) {
+	// return readAllGroups(x.db.Query("SELECT id,name,permlist FROM authgroup"))
+	rows, queryError := x.db.Query("SELECT id,name,permlist FROM authgroup")
+	if queryError != nil {
+		return nil, queryError
+	}
+	defer rows.Close()
+
+	var groups []RawAuthGroup
+	for rows.Next() {
+		name := ""
+		permList := ""
+		var id GroupIDU32
+		if errScan := rows.Scan(&id, &name, &permList); errScan != nil {
+			return nil, errScan
+		}
+		groups = append(groups, RawAuthGroup{ID: id, Name: name, PermList: permList})
```

Comment:
So for this you can just create the `RawAuthGroup` and then use it for the scan. Like
```go
r = RawAuthGroup{}
Scan(&r.id, &r.name, r.permList)
```
Makes things a bit simpler.